src/window.c

```diff
@@ -955,6 +955,8 @@ win_split_ins(
     // Do not redraw here, curwin->w_buffer may be invalid.
     ++RedrawingDisabled;
 
+    apply_autocmds(EVENT_WINNEWPRE, NULL, NULL, FALSE, NULL);
```

Comment:
What will happen if the WINNEWPRE autocmd handler creates a new window or modifies the window layout?
Is the window layout locked here?

Reply:
or when it closes other windows?

Reply:
>Is the window layout locked here?

I think locking the layout may limit the use of `WinNewPre`. I pushed the change, so the event is non-recursive now.

> What will happen if the WINNEWPRE autocmd handler creates a new window or modifies the window layout?

> or when it closes other windows?

Modifying layout should be possible and allowed. I added two new tests.

Reply:
> Modifying layout should be possible and allowed. I added two new tests.

We had in the past a lot of crashes, because the curwin or curbuf pointer became invalid, after the autocommands did something unexpected.

Reply:
I made a change to lock the layout during the event.